training.py

```diff
@@ -33,7 +33,16 @@ def train(
     dl = DataLoader(dataset, shuffle=True, batch_size=batch_size, collate_fn=data.collate_fn)
     dl = utils.infinite_dl(dl)
 
+    n_gpu = torch.cuda.device_count()
```

Comment:
Hmm... I think that this is ok but it would be better to have this as one of the input parameters with a default value of "cpu".

Where you planning to use this script with `CUDA_VISIBLE_DEVICES="specific_gpu"`?

Reply:
Incorporated your suggestion.